src/main/java/wellnus/storage/ReflectionTokenizer.java

```diff
@@ -0,0 +1,110 @@
+        try {
+            for (String indexString: splittedString) {
+                int index = Integer.parseInt(indexString);
+                cleanIndexes.add(index);
+            }
+        } catch (NumberFormatException numberFormatException) {
+            throw new TokenizerException(DETOKENIZE_ERROR_MESSAGE);
+        }
+        return cleanIndexes;
+    }
+    /**
+     * Tokenize ArrayList of Set of Integers into strings that can be stored. <br>
+     * ArrayList contains 2 Set of Integers, which corresponds for set of like indexes for the first entry
+     *      and set of pref indexes for second entry.<br>
+     * Each index will be tokenized with the following format:
+     * like [list of comma separated index] <br>
+     * pref [list of comma separated index] <br>
+     *
+     * @param arrayIndexToTokenize ArrayList that contains set of like indexes for the first entry
+     *      and set of pref indexes for the second entry. <br>
+     * @return ArrayList of Strings representing the tokenized like indexes and pref indexes that we can
+     *      write to storage.
+     */
+    public ArrayList<String> tokenize(ArrayList<Set<Integer>> arrayIndexToTokenize) throws TokenizerException {
+        ArrayList<String> tokenizedIndex = new ArrayList<>();
+        Set<Integer> likeIndexToTokenize = arrayIndexToTokenize.get(INDEX_ZERO);
+        Set<Integer> prefIndexToTokenize = arrayIndexToTokenize.get(INDEX_ONE);
+        String tokenizedLike = LIKE_KEY + " ";
+        for (int indexLike: likeIndexToTokenize) {
+            tokenizedLike = tokenizedLike + indexLike + INDEX_DELIMITER;
+        }
+        tokenizedLike = tokenizedLike.substring(INDEX_ZERO, tokenizedLike.length() - INDEX_ONE);
+        String tokenizedPref = PREF_KEY + " ";
+        for (int indexPref: prefIndexToTokenize) {
+            tokenizedPref = tokenizedPref + indexPref + INDEX_DELIMITER;
+        }
+        tokenizedPref = tokenizedPref.substring(INDEX_ZERO, tokenizedPref.length() - INDEX_ONE);
+        tokenizedIndex.add(tokenizedLike);
+        tokenizedIndex.add(tokenizedPref);
+        return tokenizedIndex;
+    }
+    /**
+     * Convert strings of tokenized Indexes into ArrayList that contains set of like indexes for the first entry
+     *      and set of pref indexes for the second entry. <br>
+     * This method can be called in the constructor of ReflectionManager to detokenize.
+     * ArrayList of indexes from storage. <br>
+     *
+     * @param tokenizedIndex List of tokenized like and pref indexes from the storage.
+     * @return ArrayList that contains set of like indexes for the first entry
+     *      and set of pref indexes for the second entry <br>
+     * @throws TokenizerException when the data can't be detokenized.
+     */
+    public ArrayList<Set<Integer>> detokenize(ArrayList<String> tokenizedIndex) throws TokenizerException {
+        ArrayList<Set<Integer>> detokenizedIndex = new ArrayList<>();
```

Comment:
I recommend something like `detokenizedIndexes`, since this is an ArrayList(plural data structure)